File: Data Structures & Algorithms/find-duplicate-integer/submission-9.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
    int firstpointer = 0;
    int secpointer = 0;
    int n = nums.size();
    while(true)
    {
        firstpointer++;
        secpointer +=2;
        if (secpointer == firstpointer)
        {
            secpointer ++;
        }
        
        if (secpointer >= n)
        {
            secpointer = secpointer - n;
        }
        if (firstpointer >= n)
        {
            firstpointer = firstpointer - n;
        }
        if (firstpointer != secpointer && nums[firstpointer] == nums[secpointer])
        {
            break;
        }


    }
    return nums[firstpointer];
    }
    
};

File: Data Structures & Algorithms/move-zeroes/submission-9.py

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        fastcur = 1
        slowcur = 0
        while fastcur < len(nums):
            if (nums[fastcur] != 0):
                if (nums[slowcur] == 0):
                    nums[slowcur],nums[fastcur] = nums[fastcur],nums[slowcur]
                    fastcur += 1
                    slowcur += 1
                else:
                    if slowcur == fastcur -1:
                        slowcur += 1
                        fastcur += 1
                    else:
                        slowcur += 1

            else:
                fastcur += 1

                    
        



        
            
            


        

        